docs/common/credentials.rst

```diff
@@ -0,0 +1,199 @@
+Credentials
+-----------
+
+This section describes the supported values for the ``--credentials`` command line argument.
+
+Expected format
+^^^^^^^^^^^^^^^
+
+Overall format
+""""""""""""""
+Each credential file must be in the following format:
+
+.. code-block:: json
+
+  {
+    "marketplace_account": "CLOUD_ALIAS",
+    "auth":
+    {
+        "KEY": "VALUE", // ...
+    }
+  }
+
+
+The supported values for the key ``marketplace_account`` are described below:
+
+Marketplace accounts (push)
+"""""""""""""""""""""""""""
+Required when using the "marketplace" or "combined" workflow
+
+- ``aws-na``
+- ``aws-emea``
+- ``azure-na``
+- ``azure-emea``
+
+Storage accounts (community-push)
+"""""""""""""""""""""""""""""""""
+Required when using the "community" or "combined" workflow
+
+- ``aws-us-storage``
+- ``aws-us-gov-storage``
+- ``aws-china-storage``
+
+Auth format
+"""""""""""
+
+The *KEY*s and *VALUE*s for the ``auth`` object are specific for each Cloud Provider implementation.
+
+**AWS Format**
+
+The following properties are supported for AWS ``auth``:
+
+- ``AWS_IMAGE_ACCESS_KEY``: The AWS access key for uploading the images and import them as AMIs
+- ``AWS_IMAGE_SECRET_ACCESS``: The AWS secret key for uploading the images and import them as AMIs
+- ``AWS_MARKETPLACE_ACCESS_KEY``: The AWS access key for publishing a new product version with its AMI into the marketplace
+- ``AWS_MARKETPLACE_SECRET_ACCESS``: The AWS secret key for publishing a new product version with its AMI into the marketplace
+- ``AWS_ACCESS_ROLE_ARN``: The AWS role ARN for publishing a new product version with its AMI into the marketplace
+- ``AWS_GROUPS``: The default sharing accounts for uploading and importing AMIs
+- ``AWS_SNAPSHOT_ACCOUNTS``: The default sharing snapshot accounts for uploading and importing AMIs
+- ``AWS_REGION``: The default region for uploading and importing AMIs
+- ``AWS_S3_BUCKET``: The S3 bucket name to upload the raw VM images
```

Comment:
@jajreidy please let me know if the description for each of these are correct.. I did them based on what I remember